app/controllers/nodes_controller.rb

```diff
@@ -22,17 +22,17 @@ def show
   # assigned role, assign a random role to it, and then call the salt
   # orchestration.
   def bootstrap
-    available_roles = [:master]
-    Minion.where(role: nil).find_each do |minion|
-      random_role = if available_roles.blank?
-        :minion
-      else
-        available_roles.delete_at(rand(available_roles.length))
-      end
-      minion.assign_role(role: random_role)
+    if Minion.where(role: nil).count > 1
+      Minion.assign_roles(roles: [:master], default_role: :minion)
+      Pharos::Salt.orchestrate
+
+      redirect_to nodes_path
+    else
+      redirect_back(
```

Comment:
well, that's "back". I just don't want to hardcode it in case we add this button elsewhere. I will change the default though from root_path to node_path.